### PR TITLE
Better way of registering builtin functions

### PR DESCRIPTION
**Pull request by kustosz**
### Pull Request Description
Given we wan't to introduce more and more builtin functions and operators, it was about time to start properly handling builtin methods.
This cleans up the way builtin atoms were defined so far and makes the primitive print operator into a builtin function.

### Important Notes
Not too many surprises here.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md), [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: Interpreter/src/main/java/org/enso/interpreter/runtime/Builtins.java [hunk 1]
```diff
@@ -1,24 +1,55 @@
...
+
+    scope.registerMethod(io, "println", PrintNode.toFunction(language));
+  }
+
+  /**
+   * Returns the {@code Unit} atom constructor.
+   *
+   * @return the {@code Unit} atom constructor
+   */
+  public AtomConstructor getUnit() {
```
Comment (iamrecursion) [inline Interpreter/src/main/java/org/enso/interpreter/runtime/Builtins.java]: I know it's not convention, but I feel like `Builtins.unit()` reads much more nicely in code than `Builtins.getUnit()`.

File: Interpreter/src/main/java/org/enso/interpreter/runtime/callable/function/ArgumentSchema.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public ArgumentSchema(
    *
    * @param argumentInfos Definition site arguments information
    */
-  public ArgumentSchema(ArgumentDefinition[] argumentInfos) {
+  public ArgumentSchema(ArgumentDefinition... argumentInfos) {
```
Comment (iamrecursion) [inline Interpreter/src/main/java/org/enso/interpreter/runtime/callable/function/ArgumentSchema.java]: Why this change?

Comment (kustosz) [reply Interpreter/src/main/java/org/enso/interpreter/runtime/callable/function/ArgumentSchema.java]: Because I'm now constructing these objects in code (in the Builtins class) and I don't feel like writing ```new ArgumentSchema(new Argument[] { foo, bar })``` every time, when I could as well write ```new ArgumentSchema(foo, bar)```
